docs/concepts/unions.md

```diff
@@ -133,6 +172,10 @@ print(user_03_uuid.int)
 
     See more details in [Required fields](../concepts/models.md#required-fields).
 
+!!! note
+    We reserve the right to tweak the internal `smart` matching algorithm in future versions of Pydantic. If you rely on very specific
+    matching behavior, it's recommended to use `union_mode='left_to_right'` or [discriminated unions](#discriminated-unions).
```

Comment:
I think this note should be next to the "Smart Mode Algorithm" box. Either move the box down here, or the note up.